src/hobbits-plugins/importerexporters/UsbDevice/usbdeviceimporteditor.cpp

```diff
@@ -0,0 +1,387 @@
+ * by functions below.
+ * 
+ * @return Parameters - the various parameters from the users UI selection, and other parameters determined by the program. 
+ */
+Parameters USBDeviceImportEditor::parameters()
+{
+    auto params = m_paramHelper->getParametersFromUi(); //gather our parameter values from the ui
+    params.insert("TransferType", (int)m_transferType); // adding the transfer type parameter
+    params.insert("TransferSize", (int)m_transferSize); // adding the transfer size parameter
+    params.insert("DeviceName", m_device);
+    return params;
+}
+
+/**
+ * @brief apply the parameters provided to the UI elements.
+ * 
+ * Applies any parameters values provided through the function parameter, to the relevant UI
+ * 
+ * @param parameters parameters and their UI elements to apply to 
+ * @return bool - returns true if successful, false if unsucessful
+ * 
+ */
+bool USBDeviceImportEditor::setParameters(const Parameters &parameters)
+{
+    return m_paramHelper->applyParametersToUi(parameters);
+}
+
+/**
+ * @brief initialize the libusb session and get the device list so we can populate the UI.
+ * 
+ * Initializes a libusb session and gets a device list so that we can populate the device, interface,
+ * alt set, and endpoint selectors. Throws errors if libusb raises an error in initalization or getting the device list.
+ * 
+ */
+void USBDeviceImportEditor::initLibusb(){
+    m_devices.clear(); //clearing the device list
+    int r = libusb_init(&m_ctx); //initialize a library session
+	if(r < 0) { //check if error, if there is an error r will be less than 0
+        std::runtime_error libusb_init_error("Error while trying to initialize Libusb");
+		throw libusb_init_error;
+	}
+    m_cnt = libusb_get_device_list(m_ctx, &m_devs);
+    if (m_cnt < 0){ //check if there is an error, if there is an error m_cnt will be less than 0
+        std::runtime_error libusb_device_list_error("Error getting device list");
+        throw libusb_device_list_error;
+    }
+}
+
+/**
+ * @brief Get a list of strings describing devices to populate the device selector.
+ * 
+ * Creates and returns a list of strings describing the various USB devices recognized by libusb, 
+ * the string contains the device number, the vendor and product ID, and the vendor and product names.
+ * 
+ * @return QStringList - list of strings describing devices recognized by libusb, used to populate the device selector 
+ */  
+QStringList USBDeviceImportEditor::getUsbDevices(){
+
+    QStringList devices; //the QStringList we will return
+    for(int i = 0; i < m_cnt; i++){ //for every device in the libusb device list
+        libusb_device *dev = m_devs[i]; //get the device
+        libusb_device_descriptor desc; 
+        libusb_get_device_descriptor(dev, &desc); //get the device descriptor
+
+        std::stringstream sstream; //set up a string stream to allow for converting a short int to hexadecimal
+        sstream << std::hex << desc.idVendor; //convert short to hex
```

Comment:
https://doc.qt.io/qt-5/qstring.html#arg-7 - this can replace both the conversion and the right-align/0-padding